src/Plugin.js

```diff
@@ -30,21 +28,24 @@ export default class Plugin extends EventEmitter {
         }
     };
 
-    constructor() {
+    constructor(emitter) {
         if (new.target === Plugin) {
             throw new TypeError("Cannot construct Plugin instances directly!");
         }
-        super();
 
         this.log = Logger.get(this.plugin.name);
+        this.interface = emitter;
+
+        this.init();
 
         /*if(this.plugin.needs.database)
             this.database = DatabaseWrapper(this.plugin.name);*/
-
-        if(typeof this.onText == 'function')
-            this.on("text", (...args) => this.onText(...args));
```

Comment:
I quite like this kind of thing, singe I want to avoid as much as possible stuff like src/plugins/Ping.js:Line#14 , not because it is not good, just because I want plugins to be as easy as possible for non-serious-programmers.

Still I think it is an improvement having the `this.interfce`, I really like this method!